pkg/reconciler/v1alpha1/taskrun/resources/input_resources.go

```diff
@@ -146,6 +157,33 @@ func AddInputResource(
 	return taskSpec, nil
 }
 
+func addVolumeFetchStep(kubeClient kubernetes.Interface, taskRun *v1alpha1.TaskRun, volumeResource *v1alpha1.VolumeResource, sourcePaths []string) ([]corev1.Container, []corev1.Volume, error) {
```

Comment:
would be great to have unit tests for `addVolumeFetchStep` and `addVolumeUploadStep` (in fact i think it would also probably make a bit more sense if we put functions like this in the same package/file with the associated resources if we can? e.g. functions inside of pkg/apis/pipeline/v1alpha1/volume_resource.go